### DISCUSSION TITLE
When using a Proxy

### DISCUSSION THREAD

**Discussion by STRATZ-Ken** [curl/curl#13834] [General]
So I was doing some debug programs. I noticed that when you set a Proxy/Host and try to go to google.com it creates a socket.  However, when I use the same Client and try to go to Yahoo.com it creates another socket.  Since the DNS lookup and transfer is done on the proxy, cant we eliminate the need for a second socket and just re-use the existing socket to the same proxy?

**Comment by icing** [comment]
Ideally curl would do that, agreed. However this is a restriction in the internal architecture that is not easily overcome.

Maybe we should add that to our TODO documentation.

**Comment by bagder** [comment]
When doing HTTPS to the server, curl just tunnels through the proxy. Such a tunnel cannot be reused against another host, it needs to create a new separate tunnel.

If you do plain HTTP, they can reuse the same single proxy connection.

**Comment by STRATZ-Ken** [comment]
Daniel,

Would this apply when you have disable ssl verification on?

Sent from my iPhone

On May 31, 2024, at 3:07 AM, Daniel Stenberg ***@***.***> wrote:

﻿

When doing HTTPS to the server, curl just tunnels through the proxy. Such a tunnel cannot be reused against another host, it needs to create a new separate tunnel.

If you do plain HTTP, they can reuse the same single proxy connection.

**Reply by bagder** [reply]
It is entirely independent of that option.
